packages/pay/src/payment-pointer.ts

```diff
@@ -0,0 +1,98 @@
+export const createHttpsUrl = (rawUrl: string, base?: string): URL | undefined => {
+  try {
+    const url = new URL(rawUrl, base)
+    if (url.protocol === 'https:') {
+      return url
+    }
+  } catch (_) {
+    return
+  }
+}
+
+/** URL of a unique account payable over Interledger, queryable via SPSP or Open Payments */
+export class AccountUrl {
+  private static DEFAULT_PATH = '/.well-known/pay'
+
+  /** Domain name of the URL */
+  private hostname: string
+
+  /** Path with stripped trailing slash, or `undefined` for default, well-known account path */
+  private path?: string
+
+  /** Query string and/or fragment. Empty string for PP, optional for the full URL format */
+  private suffix: string
+
+  /** Parse a [payment pointer](https://paymentpoiners.org) prefixed with "$" */
+  static fromPaymentPointer(paymentPointer: string): AccountUrl | undefined {
+    if (!paymentPointer.startsWith('$')) {
+      return
+    }
+
+    /**
+     * From paymentpointers.org/syntax-resolution/:
+     *
+     * "...the Payment Pointer syntax only supports a host which excludes the userinfo and port.
+     * The Payment Pointer syntax also excludes the query and fragment parts that are allowed in the URL syntax.
+     *
+     * Payment Pointers that do not meet the limited syntax of this profile MUST be
+     * considered invalid and should not be used to resolve a URL."
+     */
+    const url = createHttpsUrl('https://' + paymentPointer.substring(1))
+    if (
+      !url || // URL was invalid
+      url.username !== '' ||
+      url.password !== '' ||
+      url.port !== '' ||
+      url.search !== '' || // No query params
+      url.hash !== '' // No fragment
+    ) {
+      return
+    }
+
+    return new AccountUrl(url)
+  }
+
+  /** Parse SPSP/Open Payments account URL. Must be HTTPS, contain no credentials, and no port. */
+  static fromUrl(rawUrl: string): AccountUrl | undefined {
+    const url = createHttpsUrl(rawUrl)
```

Comment:
I don't think we should require a raw URL to be https, at least not at this layer.

Reply:
I assume the rationale is that the spec says that all payment pointers must be https. But I agree that supporting plain http would be convenient for testing. Any suggestions on where the validation should live instead?